read.go

```diff
@@ -154,6 +159,34 @@ func (r *reader) readValue(t tag.Tag, vr string, vl uint32, isImplicit bool, d *
 	}
 }
 
+// readUndefinedLengthByteValue reads a value of type OB or OW of undefined length.
+// see https://github.com/suyashkumar/dicom/issues/349 for details.
+// pixel data should not use this and instead use readPixelData.
+func (r *reader) readUndefinedLengthByteValue(t tag.Tag, vr string) (Value, error) {
+	var allData []byte
+	foundDelimeter := false
+
+	for !r.rawReader.IsLimitExhausted() {
+		data, terminated, err := r.readRawItem(false)
+		if err != nil {
+			return nil, fmt.Errorf("error reading undefined byte value: %w", err)
+		}
+
+		if terminated {
+			foundDelimeter = true
+			break
+		}
+
+		allData = append(allData, data...)
+	}
+
+	if !foundDelimeter {
+		return nil, ErrorExpectedSequenceDelimitation
+	}
```

Comment:
I'm on the fence as to whether to consider this an error here. On one hand, the spec does say that these values will be terminated by a sequence delimitation item. On the other, I know that sometimes sequences at the end of a file will drop the last sequence delimitation item. Happy to remove if you want.